src/lib/core/CHIPConfig.h

```diff
@@ -2797,6 +2797,26 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #ifndef CHIP_CONFIG_ENABLE_SERVER_IM_EVENT
 #define CHIP_CONFIG_ENABLE_SERVER_IM_EVENT 1
 #endif
+
+/**
+ * @def CHIP_CONFIG_MAX_ATTRIBUTE_STORE_ELEMENT_SIZE
+ *
+ * @brief Safety limit to ensure that we don't end up with a
+ * larger-than-expected buffer for temporary attribute storage (on the stack or
+ * in .bss).  The SDK will fail to compile if this value is set below the value
+ * it thinks it needs for a buffer size that can store any simple (not list or
+ * struct) attribute value.
+ */
+#ifndef CHIP_CONFIG_MAX_ATTRIBUTE_STORE_ELEMENT_SIZE
+// Basic Information has a ProductURL attribute that we seem to have enabled
+// everywhere, which is 256 octets.  That leads to a 259-octet buffer size.
```

Comment:
We should have a BasicInformationDataProvider interface that avoids the need to ever maintain RAM for all those attributes that are seldom read and likely backed by a constant anyway

Reply:
I agree, but that's out of scope for this PR.  We should absolutely tighten up this number by moving some of these things to External storage with access interface implementations, and reduce the RAM use in the process.

Reply:
Removed that comment anyway, because in practice that's not the limiting factor.